src/controllers/like.js

```diff
@@ -0,0 +1,44 @@
+import isUUID from 'validator/lib/isUUID';
+import { Like } from '../models';
+import { responseFormat, errorResponseFormat } from '../utils';
+
+const likeArticle = (req, res) => {
+  const { id: userId } = req.user;
+  const { articleid: articleId } = req.params;
+  const condition = { userId, articleId };
+  if (!isUUID(articleId, '4')) {
+    return res.status(400).json(errorResponseFormat({ message: 'invalid Article Id' }));
+  }
+  Like.findOrCreate({ where: condition })
+    .then((likedArticle) => {
+      if (!likedArticle[1]) {
+        Like.destroy({ where: condition }).then((result) => {
+          if (result) {
+            Like.count({ where: { articleId } }).then(articleLikeCount => res.status(200).json(
+              responseFormat({
+                status: 'success',
+                data: { message: 'you successfully liked the Article', Like: articleLikeCount },
+              }),
+            ),
+            );
+          }
+        });
+      } else {
+        Like.count({ where: { articleId } }).then(articleLikeCount => res.status(200).json(
+          responseFormat({
+            status: 'success',
+            data: { message: 'you successfully liked the Article', Like: articleLikeCount },
+          }),
+        ),
+        );
+      }
+    })
+    .catch((error) => {
+      if (error.parent.constraint === 'likes_articleId_fkey') {
+        return res
+          .status(400)
+          .json(errorResponseFormat({ message: 'articleId is not in the database' }));
```

Comment:
Good work @ssanusi. The message can be re-phrased to be 'articleId is not in the article table' to make it more specific. I think your promises chaining would have been cleaner with async/await, you might want to look into that.

Reply:
@micah-akpan its done